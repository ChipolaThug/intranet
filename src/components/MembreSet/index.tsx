import React, { useState, memo } from "react"
import { toast } from "react-toastify"

import { AppDispatch } from "../../store"

import { sendMembreSet } from "../../store/membreSet"
import { Membre } from "../../services/membres"
import styles from "./styles.module.scss"

interface Props {
    dispatch: AppDispatch
    membre: Membre
}

const MembreSet = ({ dispatch, membre }: Props) => {
    const [prenom, setPrenom] = useState(membre.prenom)
    const [nom, setNom] = useState(membre.nom)
    const [majeur, setMajeur] = useState(membre.majeur)

    const onPrenomChanged = (e: React.ChangeEvent<HTMLInputElement>) => setPrenom(e.target.value)
    const onNomChanged = (e: React.ChangeEvent<HTMLInputElement>) => setNom(e.target.value)
    const onMajeurChanged = (e: React.ChangeEvent<HTMLInputElement>) => setMajeur(+e.target.value)

    const onSavePostClicked = () => {
        if (prenom && nom) {
            dispatch(
                sendMembreSet({
                    ...membre,
                    prenom,
                    nom,
                    majeur,
                })
            )
        } else {
            toast.warning("Il faut au moins pr√©ciser un prenom et un nom", {
                position: "top-center",
                autoClose: 6000,
                hideProgressBar: true,
                closeOnClick: true,
                pauseOnHover: true,
                draggable: true,
                progress: undefined,
            })
        }
    }
    return (
        <section className={styles.MembreList}>
            <h2>Modifier un membre</h2>
            <form>
                <label htmlFor="postPrenom">
                    Prenom:
                    <input
                        type="text"
                        id="postPrenom"
                        name="postPrenom"
                        value={prenom}
                        onChange={onPrenomChanged}
                    />
                </label>
                <label htmlFor="postNom">
                    Nom:
                    <input
                        type="text"
                        id="postNom"
                        name="postNom"
                        value={nom}
                        onChange={onNomChanged}
                    />
                </label>
                <label htmlFor="postMajeur">
                    Majeur:
                    <input
                        type="text"
                        id="postMajeur"
                        name="postMajeur"
                        value={majeur}
                        onChange={onMajeurChanged}
                    />
                </label>
                <button type="button" onClick={onSavePostClicked}>
                    Save changes
                </button>
            </form>
        </section>
    )
}
export default memo(MembreSet)
