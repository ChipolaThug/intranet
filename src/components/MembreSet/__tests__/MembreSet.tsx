/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"

import MembreSet from "../index"

describe("<SetMembre />", () => {
    it("renders", () => {
        const dispatch = jest.fn()
        const tree = render(
            <MemoryRouter>
                <MembreSet
                    dispatch={dispatch}
                    membre={{
                        id: 1,
                        nom: "Aupeix",
                        prenom: "Amélie",
                        mail: "pakouille.lakouille@yahoo.fr",
                        telephone: "0675650392",
                        photo: "images/membres/$taille/amélie_aupeix.jpg",
                        alimentation: "Végétarien",
                        majeur: 1,
                        privilege: 0,
                        actif: 0,
                        commentaire: "",
                        horodatage: "0000-00-00",
                        passe: "$2y$10$fSxY9AIuxSiEjwF.J3eXGubIxUPlobkyRrNIal8ASimSjNj4SR.9O",
                    }}
                />
            </MemoryRouter>
        ).container.firstChild

        expect(tree).toMatchSnapshot()
    })
})
