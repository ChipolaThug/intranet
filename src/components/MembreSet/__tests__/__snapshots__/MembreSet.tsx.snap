// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SetMembre /> renders 1`] = `
<section
  class="MembreList"
>
  <h2>
    Modifier un membre
  </h2>
  <form>
    <label
      for="postPrenom"
    >
      Prenom:
      <input
        id="postPrenom"
        name="postPrenom"
        type="text"
        value="AmÃ©lie"
      />
    </label>
    <label
      for="postNom"
    >
      Nom:
      <input
        id="postNom"
        name="postNom"
        type="text"
        value="Aupeix"
      />
    </label>
    <label
      for="postMajeur"
    >
      Majeur:
      <input
        id="postMajeur"
        name="postMajeur"
        type="text"
        value="1"
      />
    </label>
    <button
      type="button"
    >
      Save changes
    </button>
  </form>
</section>
`;
